File: Python/ByTier/C/2039_C1_Shohag_Loves_XOR_Easy_Version.py
# Problem: CF 2039 C1 - Shohag Loves XOR (Easy Version)
# https://codeforces.com/contest/2039/problem/C1

"""
Algorithm: Brute Force with Optimization
Time Complexity: O(x * log(m)) per test case in worst case, but often much faster due to early breaks
Space Complexity: O(1)

This solution iterates through all possible values of `i` from 1 to x-1. For each `i`, it computes `y = x ^ i` and checks if `y` is within the range [1, m]. If so, it verifies whether the XOR result (x ^ y) is a divisor of either x or y (or both). This approach relies on the fact that for a given i, the corresponding y is uniquely determined by the XOR operation.

The key insight is that since y = x ^ i, we have x ^ y = x ^ (x ^ i) = i. So the check becomes: if i divides x or y, then it's valid.
"""

for _ in range(int(input())):
    x, m = map(int, input().split())
    t = 0
    for i in range(1, x):  # Loop from 1 to x-1
        y = x ^ i  # Compute y such that x ^ y = i
        if y > m:  # Skip if y exceeds m
            continue
        # Check if x ^ y (which equals i) divides either x or y
        if y % (x ^ y) == 0 or x % (x ^ y) == 0:
            t += 1
    print(t)


# https://github.com/VaHiX/CodeForces/

File: CPP/ByRound/2039/2039_C1_Shohag_Loves_XOR_Easy_Version.cpp
// Problem: CF 2039 C1 - Shohag Loves XOR (Easy Version)
// https://codeforces.com/contest/2039/problem/C1

/*
C1. Shohag Loves XOR (Easy Version)
Time Complexity: O(x * log(m)) per test case in worst case, but optimized in practice due to early exits and small x values.
Space Complexity: O(1)

Algorithms/Techniques:
- Brute force over all possible values of p from 1 to x-1
- For each p, compute y = x ^ p
- Check if y <= m and whether either x % p == 0 or y % p == 0
- Count such valid pairs

Input:
- t: number of test cases
- For each test case: x and m (integers)

Output:
- For each test case: count of valid y values where 1 <= y <= m, x != y, and x^y is divisible by either x or y.

Example Input:
5
6 9
5 7
2 3
6 4
4 1

Example Output:
3
2
1
1
0
*/

#include <cstdio>
int main() {
  long t;
  scanf("%ld", &t);
  while (t--) {
    long x, m;
    scanf("%ld %ld", &x, &m);
    long cnt(0);
    for (long p = 1; p < x; p++) {           // Iterate through all possible divisors p
      long y = x ^ p;                        // Compute y as XOR of x and p
      if (y > m) {                           // Skip if y exceeds m
        continue;
      }
      cnt += ((x % p == 0) || (y % p == 0)); // Increment counter if p divides x or y
    }
    printf("%ld\n", cnt);
  }
}


// https://github.com/VaHiX/codeForces/